server/src/main/java/org/opensearch/plugins/NetworkPlugin.java

```diff
@@ -107,4 +107,41 @@ default Map<String, Supplier<HttpServerTransport>> getHttpTransports(
     ) {
         return Collections.emptyMap();
     }
+
+    /**
+     * Returns a map of secure {@link Transport} suppliers.
+     * See {@link org.opensearch.common.network.NetworkModule#TRANSPORT_TYPE_KEY} to configure a specific implementation.
+     */
+    default Map<String, Supplier<Transport>> getSecureTransports(
+        Settings settings,
+        ThreadPool threadPool,
+        PageCacheRecycler pageCacheRecycler,
+        CircuitBreakerService circuitBreakerService,
+        NamedWriteableRegistry namedWriteableRegistry,
+        NetworkService networkService,
+        SecureTransportSettingsProvider secureTransportSettingsProvider,
+        Tracer tracer
+    ) {
+        return Collections.emptyMap();
+    }
+
+    /**
+     * Returns a map of secure {@link HttpServerTransport} suppliers.
+     * See {@link org.opensearch.common.network.NetworkModule#HTTP_TYPE_SETTING} to configure a specific implementation.
+     */
+    default Map<String, Supplier<HttpServerTransport>> getSecureHttpTransports(
+        Settings settings,
```

Comment:
What do you think about adding all (or at least some) of these parameters to a pojo to cut down on the copy pasta around these interfaces?

Reply:
Seems like a very simaliar case to SecurityApiDependencies [1] that really help keep down duplicated code in the implemening classes, such as RolesMappingApiAction [2].

I'd say there are some components the transport systems needs, a way to pool threads, allocate large arrays (?), save data to disk, throttle accepting connections, the network service, the cluster configuration (?), and inform the tracing infra on steps its taking.

If you think its out of scope I'll create an issue to follow up since we've still got time while its experimental - or do you think we need to land this interface now?

- [1] https://github.com/opensearch-project/security/blob/633ff9b30cd34acfb81a264fc49b3d716dd89c64/src/main/java/org/opensearch/security/dlic/rest/api/SecurityApiDependencies.java#L21

- [2] https://github.com/opensearch-project/security/blob/633ff9b30cd34acfb81a264fc49b3d716dd89c64/src/main/java/org/opensearch/security/dlic/rest/api/RolesMappingApiAction.java#L51

Reply:
> If you think its out of scope I'll create an issue to follow up since we've still got time while its experimental - or do you think we need to land this interface now?

I think the changes here are expected, I've only included the most necessary APis and my primary goal with that is to try to cover all existing transports (netty4, nio, reactor-netty) at bare minimum.  The `SecurityApiDependencies` & co are looking very specific to security plugin, perhaps more need to be moved to `SecureTransportSettingsProvider`, so I think we should expect this to evolve.